components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/bulkimport/CSVUserBulkImport.java

```diff
@@ -25,120 +25,121 @@
-            int successCount = 0;
-            int failCount = 0;
-            int duplicateCount = 0;
             while (line != null && line.length > 0) {
                 String userName = line[0];
 
                 int index;
                 index = userName.indexOf(CarbonConstants.DOMAIN_SEPARATOR);
                 if (index > 0) {
                     String domainFreeName = userName.substring(index + 1);
-                    userName = UserCoreUtil.addDomainToName(domainFreeName, domain);
+                    userName = UserCoreUtil.addDomainToName(domainFreeName, userStoreDomain);
                 } else {
-                    userName = UserCoreUtil.addDomainToName(userName, domain);
+                    userName = UserCoreUtil.addDomainToName(userName, userStoreDomain);
                 }
 
                 if (userName != null && userName.trim().length() > 0) {
+                    totalCount++;
                     try {
                         if (!userStore.isExistingUser(userName)) {
                             if (line.length == 1) {
                                 userStore.addUser(userName, null, null, null, null, true);
-                                success = true;
                                 successCount++;
                             } else {
                                 try {
                                     addUserWithClaims(userName, line, userStore);
-                                    success = true;
                                     successCount++;
-                                    if (log.isDebugEnabled()){
+                                    if (log.isDebugEnabled()) {
                                         log.debug("User import successful - Username : " + userName);
                                     }
-                                }catch (UserAdminException e){
+                                } catch (UserAdminException e) {
                                     fail = true;
                                     failCount++;
-                                    lastError = e.getMessage();
+                                    errorUsersMap.put(userName, e.getMessage());
                                     log.error("User import unsuccessful - Username : " + userName + " - Error: " +
                                             e.getMessage());
                                 }
                             }
                         } else {
                             isDuplicate = true;
                             duplicateCount++;
+                            duplicateUsers.add(userName);
                             log.error("User import unsuccessful - Username : " + userName + " - Error: Duplicate user");
                         }
                     } catch (UserStoreException e) {
                         if (log.isDebugEnabled()) {
-                            log.debug(e);
+                            log.debug(e.getMessage());
```

Comment:
This is a wrong way to do debug. you need to pass the error itself. If you log like this only message will be logged. Also need to give as much information as possible in debug logs.

Reply:
Changed to the proper debug logs.